app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,48 @@
+class PostsController < ApplicationController
+    def index
+      @posts = Post.all
+    end
+  
+    def show
+      @post = Post.find(params[:id])
```

Comment:
add a before acton
before_action :set_post, only: %i[ show edit update destroy ]
add a private method below
private
# Use callbacks to share common setup or constraints between actions.
def set_post
@story = Post.find(params[:id])
end
remove the line Post.find(params[:id]) from every places